rclcpp_components/include/rclcpp_components/component_manager.hpp

```diff
@@ -131,6 +131,20 @@ class ComponentManager : public rclcpp::Node
   create_component_factory(const ComponentResource & resource);
 
 protected:
+  /// Set node options for componenet
+  /*
+   * \param parameters node parameters
+   * \param remap_rules node remap rules
+   * \param request information with the node to load
+   * \return node options
+   */
+  RCLCPP_COMPONENTS_PUBLIC
+  virtual rclcpp::NodeOptions
+  SetNodeOptions(
+    std::vector<rclcpp::Parameter> parameters,
+    std::vector<std::string> remap_rules,
+    const std::shared_ptr<LoadNode::Request> request);
```

Comment:
I like the idea of a hook to generate node options that children can override.

A couple suggestions:

1. I would rename this method to `GetNodeOptions` or `CreateNodeOptions`, since it's not really setting anything on the component manager.
2. I would only pass a single `request` argument, since the parameters and remap information is already contained there in the request.
```suggestion
  CreateNodeOptions(const std::shared_ptr<LoadNode::Request> request);
```

Reply:
I've renamed this method to `CreateNodeOptions` in 5db2a22 and removed the `parameters` and `remap_rules` arguments.